src/Core/Translation/Storage/Finder/TranslationFinder.php

```diff
@@ -155,4 +157,84 @@ private function getDomainFromFile(SplFileInfo $file, string $locale): string
 
         return $domain;
     }
+
+    /**
+     * Get metadata from original catalogue, parse them to extract filename and line and add them to the updated catalogue
+     *
+     * @param MessageCatalogue $catalogue
+     * @param MessageCatalogue $originalCatalogue
+     *
+     * @return MessageCatalogue
+     */
+    private function updateCatalogueMetadata(MessageCatalogue $catalogue, MessageCatalogue $originalCatalogue): MessageCatalogue
+    {
+        $locale = $originalCatalogue->getLocale();
+        $localeSuffix = '.' . $locale;
+        $suffixLength = strlen($localeSuffix);
+
+        foreach ($originalCatalogue->all() as $domain => $messages) {
+            $originalDomain = $domain;
+            // Remove locale suffix from domain name
+            if (substr($domain, -$suffixLength) === $localeSuffix) {
+                $domain = substr($domain, 0, -$suffixLength);
+            }
+            foreach (array_keys($messages) as $translationKey) {
+                $metadata = $originalCatalogue->getMetadata($translationKey, $originalDomain);
+                if ($this->shouldAddFileMetadata($metadata) && $this->metadataContainNotes($metadata)) {
+                    $catalogue->setMetadata($translationKey, $this->parseMetadataNotes($metadata), $domain);
+                }
+            }
+        }
+
+        return $catalogue;
+    }
+
+    /**
+     * @param array|null $metadata
+     *
+     * @return bool
+     */
+    private function metadataContainNotes(array $metadata = null): bool
+    {
+        return is_array($metadata)
```

Comment:
```suggestion
        return isset($metadata['notes'][0]['content']);
```

Null content shouldn't be interresting